WIP - MTA-4637: Pattern matching docs need urgent improvements (#92)
migtools/mta-documentation: docs/topics/yaml-go-provider.adoc

// Module included in the following assemblies:
//
// * docs/rules-development-guide/master.adoc

:_mod-docs-content-type: REFERENCE
[id="yaml-go-provider_{context}"]
= Go provider

The `go` provider analyzes Go source code. This provider's capabilities are `referenced` and `dependency`.

.`referenced`

> The `referenced` capability enables the provider to find references in the source code.

Reviewer: ```suggestion
By using the `referenced` capability, the provider finds references in the source code.
```

[source,yaml]
----
when:
  go.referenced: "<regex_to_find_reference>"
----

.`dependency`

> The `dependency` capability  enables the provider to find dependencies for an application.

Reviewer: ```suggestion
By using the `dependency` capability, the provider finds dependencies for an application.
```
